ext/dom/php_dom.c

```diff
@@ -584,7 +584,7 @@ static zend_object *dom_objects_store_clone_obj(zend_object *zobject) /* {{{ */
 
 	if (instanceof_function(intern->std.ce, dom_node_class_entry) || instanceof_function(intern->std.ce, dom_modern_node_class_entry)) {
 		xmlNodePtr node = (xmlNodePtr)dom_object_get_node(intern);
-		if (node != NULL) {
+		if (node != NULL && node->doc) {
```

Comment:
The patch isn't right unfortunately.
This will break the following code: https://3v4l.org/dq0ct

This could've been revealed if your test was more elaborate, when dealing with things like this I recommend also dumping some state of the objects to see if this has consequences.

The right way to fix this is by adding a null check (`ns_mapper != NULL`) around the assignment of `clone->document->private_data`. The reason being that:
- `clone->document` will be NULL for documentless nodes, and for old DOM setting this is pointless anyway
- only new DOM uses the ns_mapper, and it will always have an internal document reference set (the fact that your `->document` or `->doc` pointer can be NULL in old DOM is imo a design mistake that I got rid of in new DOM).